### PR TITLE
[push_pull agent] Potential fix for #5713

### PR DESCRIPTION
**Pull request by sriyerg**
The first 2 commits are very minor cosmetic changes (no functional change). 

In the 3rd commit, I reverted the change made to the driver in #5889 to expose the issue outlined in #5713. I ran a test at this point to reproduce the bug by running:
```console
$ ./util/dvsim/dvsim.py hw/ip/entropy_src/dv/entropy_src_sim_cfg.hjson -i entropy_src_rng --waves --fixed-seed 1 --waves fsdb --local -v h -t xcelium
```
Rightly so, the first transaction does indeed get dropped.

Here's the snippet of the output:
```
UVM_INFO @   2327019 ps: (push_pull_host_driver.sv:36) uvm_test_top.env.m_rng_agent.driver [uvm_test_top.env.m_rng_agent.driver] rcvd item:
h_data = 0xb d_data = 0x7 host_delay = 0x0 device_delay = 0x85fa20fc 
UVM_INFO @   2327019 ps: (push_pull_host_seq.sv:38) uvm_test_top.env.m_rng_agent.sequencer [uvm_test_top.env.m_rng_agent.sequencer.m_rng_push_seq] Received response: h_data = 0xb d_data = 0x7 host_delay = 0x0 device_delay = 0x85fa20fc 
UVM_INFO @   2327019 ps: (push_pull_host_seq.sv:34) uvm_test_top.env.m_rng_agent.sequencer [uvm_test_top.env.m_rng_agent.sequencer.m_rng_push_seq] Starting sequence item: h_data = 0x5 d_data = 0xa host_delay = 0x0 device_delay = 0xd6efba2f 
UVM_INFO @   2327019 ps: (push_pull_host_driver.sv:36) uvm_test_top.env.m_rng_agent.driver [uvm_test_top.env.m_rng_agent.driver] rcvd item:
h_data = 0x5 d_data = 0xa host_delay = 0x0 device_delay = 0xd6efba2f 
UVM_INFO @   2347852 ps: (push_pull_monitor.sv:70) uvm_test_top.env.m_rng_agent.monitor [uvm_test_top.env.m_rng_agent.monitor] [] transaction detected: h_data[0x5], d_data[0x0]
UVM_INFO @   2347852 ps: (push_pull_host_seq.sv:38) uvm_test_top.env.m_rng_agent.sequencer [uvm_test_top.env.m_rng_agent.sequencer.m_rng_push_seq] Received response: h_data = 0x5 d_data = 0xa host_delay = 0x0 device_delay = 0xd6efba2f 
```
The first transaction detected by the monitor is with `h_data = 0x5`, but the first transaction sent by the driver is with `h_data = 0xb`, which gets dropped. This is evident in the waves as well:

image

In the 4th commit, I updated the `wait_clks` to wait on clocking blocks instead of absolute clk edge delays, and reran the same command as above. This time, the first transaction is applied to the DUT correctly. 

Here's the snippet of the output:
```
UVM_INFO @   2306186 ps: (push_pull_host_driver.sv:36) uvm_test_top.env.m_rng_agent.driver [uvm_test_top.env.m_rng_agent.driver] rcvd item:
h_data = 0xb d_data = 0x7 host_delay = 0x0 device_delay = 0x85fa20fc 
UVM_INFO @   2327019 ps: (push_pull_monitor.sv:75) uvm_test_top.env.m_rng_agent.monitor [uvm_test_top.env.m_rng_agent.monitor] [] transaction detected: h_data[0xb], d_data[0x0]
UVM_INFO @   2327019 ps: (push_pull_host_seq.sv:38) uvm_test_top.env.m_rng_agent.sequencer [uvm_test_top.env.m_rng_agent.sequencer.m_rng_push_seq] Received response: h_data = 0xb d_data = 0x7 host_delay = 0x0 device_delay = 0x85fa20fc 
UVM_INFO @   2327019 ps: (push_pull_host_seq.sv:34) uvm_test_top.env.m_rng_agent.sequencer [uvm_test_top.env.m_rng_agent.sequencer.m_rng_push_seq] Starting sequence item: h_data = 0x5 d_data = 0xa host_delay = 0x0 device_delay = 0xd6efba2f 
UVM_INFO @   2327019 ps: (push_pull_host_driver.sv:36) uvm_test_top.env.m_rng_agent.driver [uvm_test_top.env.m_rng_agent.driver] rcvd item:
h_data = 0x5 d_data = 0xa host_delay = 0x0 device_delay = 0xd6efba2f 
UVM_INFO @   2347852 ps: (push_pull_monitor.sv:75) uvm_test_top.env.m_rng_agent.monitor [uvm_test_top.env.m_rng_agent.monitor] [] transaction detected: h_data[0x5], d_data[0x0]
UVM_INFO @   2347852 ps: (push_pull_host_seq.sv:38) uvm_test_top.env.m_rng_agent.sequencer [uvm_test_top.env.m_rng_agent.sequencer.m_rng_push_seq] Received response: h_data = 0x5 d_data = 0xa host_delay = 0x0 device_delay = 0xd6efba2f 
``` 
The first transaction detected by the monitor is now `h_data = 0xb`, which is the same as the first transaction sent by the driver. Here's the waveform snippet:

image

In conclusion, mixing clocking block based delay statements with absolute clock edge delay statements in the same process that is driving the clocking block signals will lead to race conditions and must be avoided. 

We should probably add this to the style guide, because it is very easy to make this mistake. 

I should note that the test did fail yet again, even though it seemed to have solved the problem outlined in #5713. Here's the error message:
```
UVM_FATAL @ 10004326987 ps: (dv_utils_pkg.sv:128) [csr_utils::csr_spinwait] timeout entropy_src_reg_block.intr_state.es_entropy_valid (addr=0x752a7100) == 0x1
```
I am hoping this is a separate unrelated issue.

### CODE REVIEW

**Review by senelson7 — COMMENTED** [review]
LGTM if it fixes the #5713 issue. I'll have to simulate in all my testbenches to be sure but this PR is still valid to get rid of mixing clk/cb. Still not sure about try_next_item vs. get_next_item but that can be a separate issue which we can discuss at the next DV meeting.

**Review by rasmus-madsen — APPROVED** [review]
lgtm

**Review by udinator — APPROVED** [review]
LGTM, thanks Sri for the fix!

**Review by cindychip — APPROVED** [review]
LGTM! Thanks for the fix.

File: hw/dv/sv/push_pull_agent/push_pull_agent.sv [hunk 1]
```diff
@@ -2,16 +2,15 @@
 // Licensed under the Apache License, Version 2.0, see LICENSE for details.
 // SPDX-License-Identifier: Apache-2.0
 
-class push_pull_agent #(parameter int HostDataWidth = 32,
-                        parameter int DeviceDataWidth = HostDataWidth)
-  extends dv_base_agent #(
-  .CFG_T          (push_pull_agent_cfg#(HostDataWidth, DeviceDataWidth)),
-  .DRIVER_T       (push_pull_driver#(HostDataWidth, DeviceDataWidth)),
-  .HOST_DRIVER_T  (push_pull_host_driver#(HostDataWidth, DeviceDataWidth)),
-  .DEVICE_DRIVER_T(push_pull_device_driver#(HostDataWidth, DeviceDataWidth)),
```
Comment (sriyerg) [inline hw/dv/sv/push_pull_agent/push_pull_agent.sv:11]: Removal of these lines is the only meaningful change. Rest were made by the verible formatter.

**Review by cindychip — APPROVED** [review]
lgtm with the driver code refactor
